assembly/src/assembler/procedure_cache.rs

```diff
@@ -72,6 +72,7 @@ impl ProcedureCache {
                 if proc.num_locals() != cached_proc.num_locals() {
                     Err(AssemblyError::conflicting_num_locals(proc.name()))
                 } else {
+                    self.proc_id_map.entry(*proc.id()).or_insert(proc.mast_root());
```

Comment:
I think this could be a simple `.insert()` here as we have already checked that this procedure ID is not in this cache on line 63.

Reply:
Good catch, should we update the pattern on line 80 as well?

Reply:
Yep - let's do it.

Reply:
done